apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/site-logo/edit.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * External dependencies
+ */
+import { Component } from '@wordpress/element';
+import { InnerBlocks } from '@wordpress/editor';
+import { Placeholder, Spinner } from '@wordpress/components';
+import apiFetch from '@wordpress/api-fetch';
+
+class SiteLogoEdit extends Component {
+	state = {
+		siteLogoId: null,
+		siteLogoAttachmentUrl: null,
+		siteLogoWidth: null,
+		siteLogoHeight: null,
+		siteLogoError: null,
+	};
+
+	fromApi = function( { site_logo } ) {
+		const { id, url, width, height } = site_logo;
+		return {
+			siteLogoId: id,
+			siteLogoAttachmentUrl: url,
+			siteLogoWidth: width,
+			siteLogoHeight: height,
+		};
+	};
+
+	componentDidMount() {
+		apiFetch( { path: '/full-site-editing/v1/site-logo' } ).then(
+			data => {
+				this.setState( this.fromApi( data ) );
+			},
+			error => {
```

Comment:
In most promise implementations the second argument is the rejection handler. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then

apiFetch().then().catch() is a little different since catch will also fire when an exception is thrown from the promise success handler.